src/module_4/train.py

```diff
@@ -0,0 +1,126 @@
+import pandas as pd
+import numpy as np
+import logging
+import os
+import matplotlib
+import matplotlib.pyplot as plt
+from typing import Tuple
+from sklearn.metrics import precision_recall_curve, auc, roc_auc_score, roc_curve
+from utils import *
+from sklearn.base import BaseEstimator
+import datetime
+import joblib
+import xgboost as xgb 
+import json
+
+logger = logging.getLogger(__name__)
+logger.level = logging.INFO
+
+consoleHandler = logging.StreamHandler()
+logger.addHandler(consoleHandler)
+
+TARGET = ['outcome']
+FEATURE_COLS = ['user_order_seq', 'ordered_before', 'abandoned_before',
+       'active_snoozed', 'set_as_regular', 'normalised_price', 'discount_pct',
+       'global_popularity', 'count_adults', 'count_children', 'count_babies',
+       'count_pets', 'people_ex_baby', 'days_since_purchase_variant_id',
+       'avg_days_to_buy_variant_id', 'std_days_to_buy_variant_id',
+       'days_since_purchase_product_type', 'avg_days_to_buy_product_type',
+       'std_days_to_buy_product_type']
+
+TRAIN_SIZE = 0.7
+
+OUTPUT_PATH = os.path.join(STORAGE_PATH, "module_4_models")
+
+PARAM_GRID = {'learning_rate': [0.001, 0.01, 0.1, 0.25, 1, 10]}
+
+
+def train_test_split(df:pd.DataFrame, train_size:float) -> Tuple[pd.DataFrame, pd.Series, pd.DataFrame, pd.Series]:
+
```

Comment:
This will benefit from a different name and or some docstrings. `train_test_split` is a very well-known sklearn method that could be confusing since you are splitting data here based on time and using quantiles.